msb/analysis/timestepinterpolator.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+from msb.network.pubsub.types import Subscriber
+
+
+class PandasInterpolator:
+    """
+    An Iterator class that interpolates data points to a fixed rate.
+    """
+
+    def __init__(self, sub: Subscriber, freq_ms: float, sample_size: int = 100):
+        self.sub = sub
+        self.incoming: pd.DataFrame = None
+        self.outgoing: pd.DataFrame = None
+        self.resampling_frequency = freq_ms
+        self.sample_size = sample_size
```

Comment:
Technically this is not a "sample", consider renaming, e.g. `batch_size` or `points_per_batch`.